modules/react/text-input/lib/InputGroup.tsx

```diff
@@ -42,7 +55,69 @@ export const InputGroupInnerEnd = createComponent('div')({
 
 export const InputGroupInput = createComponent(TextInput)({
   Component(elemProps: ExtractProps<typeof Flex>, ref, Element) {
-    return <Flex ref={ref} as={Element} width="100%" {...elemProps} />;
+    const inputRef = React.useContext(InputGroupContext);
+    const elementRef = useForkRef(ref, inputRef);
+    return <Flex ref={elementRef} as={Element} width="100%" {...elemProps} />;
+  },
+});
+
+export interface ClearInputButtonProps {}
+
+/**
+ * A clear input button. This can be a component later.
+ */
+export const ClearInputButton = createComponent(TertiaryButton)({
+  Component(elemProps: ClearInputButtonProps, ref, Element) {
+    const inputRef = React.useContext(InputGroupContext);
+    const localRef = React.useRef<HTMLButtonElement>(null);
+    const elementRef = useForkRef(localRef, ref);
+    const [inputHasValue, setInputHasValue] = React.useState(false);
+
+    React.useLayoutEffect(() => {
+      const input = inputRef.current;
+      const button = localRef.current;
+
+      if (input && button) {
+        input.addEventListener('input', () => {
+          setInputHasValue(!!input.value);
+        });
+      }
+    }, [inputRef, localRef]);
+
+    const props = mergeProps(
+      {
+        onMouseDown(event: React.MouseEvent) {
+          event.preventDefault();
+        },
+        onClick() {
+          dispatchInputEvent(inputRef.current, '');
+        },
+      },
+      elemProps
+    );
+
+    return (
+      <Element
+        ref={elementRef}
+        // This element does not need to be accessible via screen reader. The user can already clear
+        // an input
```

Comment:
so no need for an `aria-label` ?

Reply:
It is hidden from screen readers and keyboard users because of the `tabIndex={-1}` and `role="presentation"`.

This isn't specified yet, but the keyboard already has the ability to clear the input. Clearing the input is mostly for mouse and touch users. This could change in the future, however. Accessibility likes less tab stops if they are not necessary, but there may be an argument for making the button accessible via keyboard or screen reader.